controllers/chat/initChatFromPost.js

```diff
@@ -178,7 +213,7 @@ const initChatFromPost = async (req, res) => {
     await client.connectUser(user, req.token);
     // check if the channel already exists
     let newChannel;
-    let source_id = source === 'post' ? postId : commentId;
+    let source_id = postId;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change in how `source_id` is assigned might cause issues if the source is not 'post'. In the old hunk, `source_id` was assigned based on the source type, but in the new hunk, it's always assigned `postId`. If the source is not 'post', this could lead to incorrect behavior. Please ensure that this change doesn't affect other parts of the code that rely on `source_id` being the id of the comment when the source is 'comment'.

```diff
- let source_id = postId;
+ let source_id = source === 'post' ? postId : commentId;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->